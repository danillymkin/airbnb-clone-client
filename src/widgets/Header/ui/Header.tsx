import React, { FunctionComponent } from 'react';
import { Container } from '@/shared/ui/Container';
import { Logo } from '@/shared/ui/Logo';
import { Search } from '@/features/Search';
import { UserMenu } from '@/features/UserMenu';

export const Header: FunctionComponent = ({}): JSX.Element => {
  return (
    <header
      data-testid={'header'}
      className={'fixed z-10 w-full bg-white shadow-sm'}
    >
      <div className={'border-b py-4'}>
        <Container>
          <div className={'flex items-center justify-between gap-3 md:gap-0'}>
            <div className={'hidden md:block'}>
              <Logo />
            </div>

            <Search />

            <UserMenu />
          </div>
        </Container>
      </div>
    </header>
  );
};
