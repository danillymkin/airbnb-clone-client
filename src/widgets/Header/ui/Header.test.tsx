import { Header } from './Header';
import { render, screen } from '@testing-library/react';

describe('Header', () => {
  it('should render header', () => {
    render(<Header />);
    const header = screen.getByTestId('header');
    expect(header).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    render(<Header />);
    const header = screen.getByTestId('header');
    expect(header).toMatchSnapshot();
  });

  it('should render logo', () => {
    render(<Header />);
    const logo = screen.getByAltText(/logo/i);
    expect(logo).toBeInTheDocument();
  });

  it('should render search', () => {
    render(<Header />);
    const search = screen.getByRole('search');
    expect(search).toBeInTheDocument();
  });

  it('should render user menu', () => {
    render(<Header />);
    const userMenu = screen.getByTestId('user-menu');
    expect(userMenu).toBeInTheDocument();
  });
});
