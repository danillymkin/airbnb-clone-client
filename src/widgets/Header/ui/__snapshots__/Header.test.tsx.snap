// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<header
  class="fixed z-10 w-full bg-white shadow-sm"
  data-testid="header"
>
  <div
    class="border-b py-4"
  >
    <div
      class="mx-auto max-w-[2520px] px-4 sm:px-2 md:px-10 xl:px-20"
    >
      <div
        class="flex items-center justify-between gap-3 md:gap-0"
      >
        <div
          class="hidden md:block"
        >
          <img
            alt="Logo"
            class="block cursor-pointer"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="31"
            src="/_next/image?url=%2Fimages%2Flogo.png&w=256&q=75"
            srcset="/_next/image?url=%2Fimages%2Flogo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogo.png&w=256&q=75 2x"
            style="color: transparent;"
            width="100"
          />
        </div>
        <div
          class="w-full cursor-pointer rounded-full border py-2 shadow-sm transition hover:shadow-md md:w-auto"
          role="search"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="px-6 text-sm font-semibold"
            >
              Anywhere
            </div>
            <div
              class="hidden flex-1 border-x px-6 text-center text-sm font-semibold sm:block"
            >
              Any Week
            </div>
            <div
              class="flex items-center gap-3 pl-6 pr-2 text-sm text-gray-600"
            >
              <div
                class="hidden sm:block"
              >
                Add Guests
              </div>
              <div
                class="rounded-full bg-rose-500 p-2 text-white"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  role="img"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="relative"
          data-testid="user-menu"
        >
          <div
            class="flex items-center gap-3"
          >
            <div
              class="hidden cursor-pointer rounded-full px-4 py-3 text-sm font-semibold transition hover:bg-neutral-100 md:block"
            >
              Airbnb your home
            </div>
            <div
              class="relative"
              data-headlessui-state=""
              data-testid="menu"
            >
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                data-headlessui-state=""
                id="headlessui-menu-button-:r2:"
                type="button"
              >
                <div
                  class="flex cursor-pointer items-center gap-3 rounded-full border border-neutral-200 p-4 transition hover:shadow-md md:px-2 md:py-1"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
                    />
                  </svg>
                  <div
                    class="hidden md:block"
                  >
                    <img
                      alt="Avatar"
                      class="rounded-full"
                      data-nimg="1"
                      decoding="async"
                      height="30"
                      loading="lazy"
                      src="/_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=64&q=75"
                      srcset="/_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=32&q=75 1x, /_next/image?url=%2Fimages%2Favatar-fallback.jpg&w=64&q=75 2x"
                      style="color: transparent;"
                      width="30"
                    />
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
